test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -512,6 +527,364 @@ TEST(ParticipantTests, ChangePSMDomainParticipantQos)
 
 }
 
+class DomainParticipantTest : public DomainParticipant
+{
+public:
+
+    const DomainParticipantImpl* get_impl() const
+    {
+        return impl_;
+    }
+
+};
+
+void get_rtps_attributes(
+        const DomainParticipant* participant,
+        fastrtps::rtps::RTPSParticipantAttributes& att)
+{
+    const DomainParticipantTest* participant_test = static_cast<const DomainParticipantTest*>(participant);
+    ASSERT_NE(nullptr, participant_test);
+    const DomainParticipantImpl* participant_impl = participant_test->get_impl();
+    ASSERT_NE(nullptr, participant_impl);
+    att = participant_impl->rtps_participant()->getRTPSParticipantAttributes();
+}
+
+void helper_wait_for_at_least_entries(
+        const MockConsumer* mockConsumer,
+        uint32_t amount)
+{
+    const uint32_t AsyncTries = 5;
+    const uint32_t AsyncWaitMs = 25;
+    size_t entries = 0;
+    for (uint32_t i = 0; i != AsyncTries; i++)
+    {
+        entries = mockConsumer->ConsumedEntries().size();
+        if (entries >= amount)
+        {
+            break;
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(AsyncWaitMs));
+    }
+    EXPECT_LE(amount, mockConsumer->ConsumedEntries().size());
+}
+
+/**
+ * Test that checks the configuration of remote server list using both the API and the environment variable
+ */
+TEST(ParticipantTests, RemoteServersListConfiguration)
+{
+    Log::ClearConsumers();
+    MockConsumer* mockConsumer = new MockConsumer("RTPS_QOS_CHECK");
+    Log::RegisterConsumer(std::unique_ptr<LogConsumer>(mockConsumer));
+    Log::SetVerbosity(Log::Warning);
+
+    // Set environment variables
+    /**
+     * FASTDDS_ENVIRONMENT_FILE_ENV_VAR should be defined from the start because it is only loaded once (when the XML
+     * profiles are loaded). The file is not going to exist until a later case. However, the filewatch is initialized
+     * even though the file does not exist ()
```

Comment:
According to comments later in this test and this check:

https://github.com/eProsima/Fast-DDS/blob/4f6ca2a3dcdd57f0e8537b086c09df916af5ab98/src/cpp/rtps/RTPSDomain.cpp#L120

It looks like the filewatch is in fact NOT initialized if the file does not exist. If that is so please correct this comment.

Reply:
This comment does not apply any longer as the test has been split into smaller tests